### DISCUSSION TITLE
sympy Min/Max with assumptions

### DISCUSSION THREAD

**Discussion by georgebisbas** [sympy/sympy#22515] [Ideas]
Dear all at Sympy, congrats on your work!
I am trying to use sympy Min/Max to evaluate a set of variables.
I have looked at: https://docs.sympy.org/latest/modules/functions/elementary.html#min
and
https://docs.sympy.org/latest/guides/assumptions.html

e.g.
a, b, c, d (we have no knowledge for them)
sympy.Max(a,b,c,d) = Max(a,b,c,d)

In case I have the hypothesis that a>c can I somehow get
sympy.Min(a,b,c,d) = Min(a,b,d) ?
Any help appreciated,
best,
George

**Comment by oscargus** [comment]
if I understand things correctly, the idea is that one should be able to do this using `refine`, so `refine(Max(a, b, c, d), Q.gt(a, c))` should ideally give this. However, it is not in a mature enough state to handle this.

**Reply by georgebisbas** [reply]
Hi Oscar, 
thank you for your reply. This minimal example does not seem to work though:
```
a = Symbol('a')
b = Symbol('b')
c = Symbol('c')
d = Symbol('d')

print(refine(Max(a, b, c, d), Q.(gt(a, c))))
```
nor this:
```
import sympy
from sympy import refine, Symbol, Q, Max

a = Symbol('a')
b = Symbol('b')
c = Symbol('c')
d = Symbol('d')

print(refine(Max(a, b, c, d), Q.is_true(sympy.Gt(a, c))))

```

**Reply by oscargus** [reply]
You are right that it doesn't work. That was the meaning of the last sentence, although maybe overly abstract. But that is the idea of how it should work.

I actually made a PR to get it to work, #22615, but it wasn't that easy to get a good solution for it (basically the problem I encountered was that while `refine(a > c, Q.gt(a, c))` works, `refine(a >= c, Q.gt(a, c))` don't, meaning that one need to write the assumptions in a specific way to get it to work, which is not really something that should be needed.)
